### ISSUE TITLE
catkin_prepare_release fails for a protected target branch

### ISSUE THREAD

**Issue by 130s** [ros/catkin#937]
I don't know if any clean solution is available for this issue. Just describing an issue I have.

Looks like https://github.com/ros/catkin/issues/463 can happen when I run `catkin_prepare_release` from a branch where certain configuration is not yet done. And from my experience that kind of branch is a "feature branch", one that's created temporarily and will be deleted once its purpose is done.

```
$ catkin_prepare_release                                           
Prepare the source repository for a release.
Repository type: git
Found packages: pkg_a, pkg_b,...
Prepare release of version '1.0.6' [Y/n]?
<string>:12: (WARNING/2) Inline emphasis start-string without end-string.
<string>:12: (WARNING/2) Inline emphasis start-string without end-string.
Trying to push to remote repository (dry run)...
Could not determine git remote: Command '['/usr/bin/git', 'config', '--get', 'branch.sample_branch.remote']' returned non-zero exit status 1

May be the branch 'sample_branch' is not tracking a remote branch?

$ which catkin_prepare_release 
/opt/ros/kinetic/bin/catkin_prepare_release
$ apt-cache policy ros-kinetic-catkin
ros-kinetic-catkin:
  Installed: 0.7.11-0xenial-20180222-175501-0800
  Candidate: 0.7.11-0xenial-20180222-175501-0800
  Version table:
 *** 0.7.11-0xenial-20180222-175501-0800 500
        500 http://packages.ros.org/ros/ubuntu xenial/main amd64 Packages
        100 /var/lib/dpkg/status
```

Manual workaround is to run a following command as posted on https://github.com/ros/catkin/issues/463#issuecomment-20703183
```
git branch %YOUR_BRANCH% --set-upstream remotes/%YOUR_REMOTE%/%YOUR_BRANCH%
```

Btw, this may not be an issue of `catkin_prepare_release` but I'm just sharing. Reason why I run from a temporary branch is on some repos the target branch is protected and doesn't allow direct commit, which `catkin_prepare_release` wants to do. So I open a PR to the target branch and have it merged. 

A problem of this flow is that `tag` gets created based on the temporary branch's status at the time `catkin_prepare_release` succeeds, which is never the same with the status when the PR is merged. To work this around, I delete the tag that's just created both on local and on remote. Then once PR is merged I `git tag %TAG%` locally then push to the remote.

**Comment by dirk-thomas** [comment]
I am not sure I understand what you are asking for. `catkin_prepare_release` does two parts:
* modify the local versions and commit the changes locally
* push the changes as well as a tag upstream

If you local branch is not tracking a remote branch the tool won't be able to do the second step. Probably the same is true when the remote branch is protected.

Do you want an option to only perform the first step?

**Comment by dirk-thomas** [comment]
I am closing this for now due to no response. If you have additional information please create a new ticket in the [catkin_pkg](https://github.com/ros-infrastructure/catkin_pkg) repo since the functionality has been moved as of #941.
